specs/APIReview_CustomDownload.md

```diff
@@ -0,0 +1,346 @@
+
+    Example:
+
+    ```
+    /// Event args for the NewWindowRequested event. The event is fired when content
+    /// inside webview requested to open a new window (through window.open() and so on.)
+    [uuid(34acb11c-fc37-4418-9132-f9c21d1eafb9), object, pointer_default(unique)]
+    interface ICoreWebView2NewWindowRequestedEventArgs : IUnknown
+    {
+        // ...
+
+        /// Window features specified by the window.open call.
+        /// These features can be considered for positioning and sizing of
+        /// new webview windows.
+        [propget] HRESULT WindowFeatures([out, retval] ICoreWebView2WindowFeatures** windowFeatures);
+    }
+    ```
+
+    ```c# (but really MIDL3)
+    public class CoreWebView2NewWindowRequestedEventArgs
+    {
+        // ...
+
+	       public CoreWebView2WindowFeatures WindowFeatures { get; }
+    }
+    ```
+-->
+## Win32 C++
+```c#
+[uuid(9EAFB7D0-88C3-4450-BBFB-C05A46C40C72), object, pointer_default(unique)]
+interface ICoreWebView2Staging2 : IUnknown {
+
+  /// Add an event handler for the `DownloadStarting` event. This event fires
+  /// when a download has begun, blocking the default download UI.
+  ///
+  /// The host can choose to cancel a download, change the save path, and hide
+  /// the default download UI. If the host chooses to cancel the download, the
+  /// download is not saved and no UI is shown. Otherwise, the download is saved
+  /// once the event completes, and default download UI is shown if the host
+  /// did not choose to hide it.
+  ///
+  /// If the event is not handled, downloads will complete normally with the
+  /// default UI shown.
+  ///
+  HRESULT add_DownloadStarting(
+    [in] ICoreWebView2StagingDownloadStartingEventHandler* eventHandler,
+    [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_DownloadStarting`.
+  HRESULT remove_DownloadStarting(
+      [in] EventRegistrationToken token);
+}
+
+[uuid(53b676ec-c3b1-4804-996a-c72c16b09efb), object, pointer_default(unique)]
+interface ICoreWebView2StagingDownloadStartingEventHandler : IUnknown
+{
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2Staging2* sender,
+      [in] ICoreWebView2StagingDownloadStartingEventArgs* args);
+}
+
+/// Event args for the `DownloadStarting` event.
+[uuid(c7e95e2f-f789-44e6-b372-3141469240e4), object, pointer_default(unique)]
+interface ICoreWebView2StagingDownloadStartingEventArgs : IUnknown
+{
+  /// The uri of the download.
+  [propget] HRESULT Uri([out, retval] LPWSTR* uri);
+
+  /// The Content-Disposition header value from HTTP response.
+  [propget] HRESULT ContentDisposition([out, retval] LPWSTR* contentDisposition);
+
+  /// MIME type of the downloaded content.
+  [propget] HRESULT MimeType([out, retval] LPWSTR* mimeType);
+
+  /// The total number of expected bytes.
+  [propget] HRESULT TotalBytes([out, retval] UINT64* totalBytes);
```

Comment:
Bytes is the unit. This should probably say size somewhere. Looking in Tempo for examples of things that say 'bytes' here are some other options that exist:
* TotalSizeInBytes
* DownloadSizeInBytes